src/libcollections/string.rs

```diff
@@ -343,7 +343,8 @@ impl String {
     /// This function is unsafe because we dereference memory until we find the
     /// NUL character, which is not guaranteed to be present. Additionally, the
     /// slice is not checked to see whether it contains valid UTF-8
-    #[unstable = "just renamed from `mod raw`"]
+    #[deprecated = "use ffi::c_str_to_bytes + str::from_utf8 + .to_string()"]
```

Comment:
I'm somewhat worried about a large array of functions popping up in "view this C string as a rust type":

* `c_str_to_bytes` x 2 for `_with_nul`
* `c_str_to_bytes_with_max_size` (e.g. #20475) x 2 for `_with_nul`
* `c_str_to_str`

I think I'd be more in favor of sticking to the bare bones for now, and perhaps expanding later? After thinking a bit, I think that #20475 may be able to be handled by `slice::from_raw_buf` plus a search for 0, and the `to_str` variant may not actually come up all that often depending on the application. I suppose I'm always a fan of conservativeness, and we do tend to avoid providing too many convenience methods that are just compositions of some others.